tests/oom-test.nix

```diff
@@ -0,0 +1,102 @@
+{
+  self,
+  nixpkgs ? builtins.fetchTarball "https://github.com/nixOS/nixpkgs/archive/22.05.tar.gz",
+  pkgs ? import nixpkgs {},
+} : rec {
+  examplesRepo = pkgs.stdenvNoCC.mkDerivation (finalAttrs: {
+    pname = "examples";
+    version = "0";
+
+    src = pkgs.fetchFromGitHub {
+      owner = "wharfix";
+      repo = "examples";
+      rev = "c09ebe2355b69f919f0dd2197d3d4c609fb6fb33";
+      hash = "sha256-sKvmqbMvJeLv0vo7kgVBdqSuKsOFdUXsbizlrVwxjlA=";
+    };
+
+
+    dontConfigure = true;
+    dontBuild = true;
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out/
+      cp $src $out/repo -r
+
+      runHook postInstall
+    '';
+  });
+  type = "app";
+  default = program;
+  program = pkgs.nixosTest {
+    name = "docker-registry-2";
+    meta = with pkgs.lib.maintainers; {
+      maintainers = [ globin ironpinguin ];
+    };
+
+    nodes = {
+      registry = { config, ... }: {
+        virtualisation.memorySize = 4096; # 2 G
+        virtualisation.diskSize = 100240; # 10 G
+        virtualisation.writableStoreUseTmpfs = false;
+
+        environment.etc.nixpkgs.source = builtins.fetchTarball {
```

Comment:
Could reuse `nixpkgs` here from the outermost argset, instead of fetching yet another one

Reply:
IIRC we did this so wharfix/examples (that uses this same rev in `default.nix`) wouldn't have to touch the network, back when that was sandboxed.

Reply:
This was removed